docs/src/man/basics.md

```diff
@@ -0,0 +1,1220 @@
+
+julia> german."Sex", german[!, "Sex"] # the same using string indexing
+(["male", "female", "male", "male", "male", "male", "male", "male", "male", "male"  …  "male", "male", "male", "male", "male", "female", "male", "male", "male", "male"], ["male", "female", "male", "male", "male", "male", "male", "male", "male", "male"  …  "male", "male", "male", "male", "male", "female", "male", "male", "male", "male"])
+
+julia> german[:, 3] # note that this creates a copy
+1000-element PooledArrays.PooledVector{String, UInt32, Vector{UInt32}}:
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ ⋮
+ "male"
+ "male"
+ "male"
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+```
+
+To grab one row as DataFrame, we can index as follows:
+```jldoctest dataframe
+julia> german[2:2, :]
+1×10 DataFrame
+ Row │ id     Age    Sex     Job    Housing  Saving accounts  Checking account ⋯
+     │ Int64  Int64  String  Int64  String   String           String           ⋯
+─────┼──────────────────────────────────────────────────────────────────────────
+   1 │     1     22  female      2  own      little           moderate         ⋯
+                                                               3 columns omitted
+
+julia> german[3, :] # this produces a DataFrameRow which is treated as 1-dimensional object similar to a NamedTuple
+DataFrameRow
+ Row │ id     Age    Sex     Job    Housing  Saving accounts  Checking account ⋯
+     │ Int64  Int64  String  Int64  String   String           String           ⋯
+─────┼──────────────────────────────────────────────────────────────────────────
+   3 │     2     49  male        1  own      little           NA               ⋯
+                                                               3 columns omitted
+```
+
+we can grab a single cell or element with the same syntax to grab an element of an array:
+```jldoctest dataframe
+julia> german[4, 4]
+2
+```
+
+or to get a new DataFrame that is subset of rows and columns:
+```jldoctest dataframe
+julia> german[4:5, 4:5]
+2×2 DataFrame
+ Row │ Job    Housing
+     │ Int64  String
+─────┼────────────────
+   1 │     2  free
+   2 │     2  free
+```
+
+## Not, Between, Cols, and All selectors
+
+Finally, you can use `Not`, `Between`, `Cols`, and `All` selectors in more complex column selection
+scenarioes (note that `Cols()` selects no columns while `All()` selects all columns therefore `Cols`
+is a preferred selector if you write generic code). The following examples move all columns whose
+names match `r"x"` regular expression respectively to the front and the end of a dataframe. Now we will
+see how `cols` and `Between` can be used to select columns of a DataFrame.
+
+A `Not` selector (from the [InvertedIndices](https://github.com/mbauman/InvertedIndices.jl) package) can be
+used to select all columns excluding a specific subset:
+
+```jldoctest dataframe
+julia> german[!, Not(:Age)]
```

Comment:
maybe comment that `:Age` was after `:id`. Also better not use `!` - it should not be used in such cases